executor/infoschema_reader.go

```diff
@@ -2616,54 +2614,73 @@ type tiflashInstanceInfo struct {
 }
 
 func (e *TiFlashSystemTableRetriever) initialize(sctx sessionctx.Context, tiflashInstances set.StringSet) error {
-	store := sctx.GetStore()
-	if etcd, ok := store.(kv.EtcdBackend); ok {
-		var addrs []string
-		var err error
-		if addrs, err = etcd.EtcdAddrs(); err != nil {
+	storeInfo, err := infoschema.GetStoreServerInfo(sctx)
+	if err != nil {
+		return err
+	}
+
+	for _, info := range storeInfo {
+		if info.ServerType != kv.TiFlash.Name() {
+			continue
+		}
+		info.ResolveLoopBackAddr()
+		if len(tiflashInstances) > 0 && !tiflashInstances.Exist(info.Address) {
+			continue
+		}
+		hostAndStatusPort := strings.Split(info.StatusAddr, ":")
+		if len(hostAndStatusPort) != 2 {
+			return errors.Errorf("node status addr: %s format illegal", info.StatusAddr)
+		}
+		// fetch tiflash config
+		configURL := fmt.Sprintf("%s://%s/config", util.InternalHTTPSchema(), info.StatusAddr)
+		resp, err := util.InternalHTTPClient().Get(configURL)
+		if err != nil {
+			sctx.GetSessionVars().StmtCtx.AppendWarning(err)
+			continue
+		}
+		if resp.StatusCode != http.StatusOK {
+			return errors.Errorf("request %s failed: %s", configURL, resp.Status)
+		}
```

Comment:
should append warning or return error first?

Reply:
If tiflash doesn't return response, we can safely ignore the tiflash node and just append warning.
But if tiflash can return response, but the status code is not `StatusOK`, then something bad must happen and an error is better to remind it.